sotodlib/mapmaking/demod_mapmaker.py

```diff
@@ -310,13 +357,127 @@ def from_work(self, map):
         if self.tiled: return tilemap.redistribute(map, self.comm, self.rhs.geometry.active)
         else: return utils.allreduce(map, self.comm)
 
-    def write(self, prefix, tag, m):
+    def write(self, prefix, tag, m, write_partial_fits=False):
         if not self.output: return
         oname = self.ofmt.format(name=self.name)
         oname = "%s%s_%s.%s" % (prefix, oname, tag, self.ext)
-        if self.tiled:
-            tilemap.write_map(oname, m, self.comm)
+        if self.pix_scheme == "rectpix":
+            if self.tiled:
+                tilemap.write_map(oname, m, self.comm)
+            else:
+                if self.comm is None or self.comm.rank == 0:
+                    enmap.write_map(oname, m)
         else:
-            if self.comm is None or self.comm.rank == 0:
-                enmap.write_map(oname, m)
+            if self.ext in ["fits", "fits.gz"]:
+                if hp is None:
+                    raise ImportError("Cannot save healpix map as fits; healpy could not be imported. Install healpy or save as npy or h5")
+                if m.ndim > 2:
+                    m = np.reshape(m, (np.product(m.shape[:-1]), m.shape[-1]), order='C') # Flatten wrapping axes; healpy.write_map can't handle >2d array
+                hp.write_map(oname, m.view(self.dtype), nest=(self.hp_geom.ordering=='NEST'), partial=write_partial_fits, overwrite=True)
+            elif self.ext == "npy":
+                np.save(oname, m.view(self.dtype))
```

Comment:
I think we'd be wise to write both the array and some supporting info the projection.  NEST is essential for understanding the data.  NSIDE is helpful.  Eventually we might throw in the celestial frame (galactic / equatorial).

Reply:
Like this, basically:
```
np.save('test.npy', {'nside': 10, 'data': np.zeros(1000)}, allow_pickle=True)
```

Reply:
Got it, thanks for the help!